runtime/src/rpc.rs

```diff
@@ -93,37 +93,45 @@ impl PolkaBtcProvider {
             self.account_id.clone(),
             Option::<H256>::None,
         )
-        .await?;
-        log::info!("Refreshing nonce to {}", account_info.nonce);
+        .await
+        .unwrap_or_default();
+        log::info!("Refreshing nonce: {}", account_info.nonce);
         signer.set_nonce(account_info.nonce);
-        Ok(())
     }
 
     /// Gets a copy of the signer with a unique nonce
     async fn with_unique_signer<F, R, T>(&self, call: F) -> Result<T, Error>
     where
-        F: FnOnce(PolkaBtcSigner) -> R,
+        F: Fn(PolkaBtcSigner) -> R,
         R: Future<Output = Result<T, SubxtError>>,
     {
-        let signer = {
-            let mut signer = self.signer.write().await;
-            // return the current value, increment afterwards
-            let cloned_signer = signer.clone();
-            signer.increment_nonce();
-            cloned_signer
-        };
-        match call(signer).await {
-            Ok(val) => Ok(val),
-            Err(SubxtError::Rpc(RequestError::Request(JsonRpcError { error, .. })))
-                if error == JsonRpcErrorCode::ServerError(POOL_INVALID_TX).into() =>
-            {
-                // here is no way to know why the transaction is invalid, so always
-                // refresh nonce and propogate to caller
-                self.refresh_nonce().await?;
-                Err(Error::InvalidTransaction)
-            }
-            Err(err) => Err(err.into()),
-        }
+        notify_retry(
+            || async {
+                let signer = {
+                    let mut signer = self.signer.write().await;
+                    // return the current value, increment afterwards
+                    let cloned_signer = signer.clone();
+                    signer.increment_nonce();
+                    cloned_signer
+                };
+                call(signer).await
+            },
+            |result| async {
+                match result {
+                    Ok(ok) => Ok(ok),
+                    Err(SubxtError::Rpc(RequestError::Request(JsonRpcError { error, .. })))
+                        if error == JsonRpcErrorCode::ServerError(POOL_INVALID_TX).into() =>
```

Comment:
We saw there were situation where this error was thrown to other reasons than the nonce being outdated. Maybe we should retry only a limited number of times because of that